lithops/serverless/backends/gcp_functions/entry_point.py

```diff
@@ -23,20 +23,36 @@
 from lithops.utils import setup_lithops_logger
 from lithops.worker import function_handler
 from lithops.worker import function_invoker
+from lithops.worker.utils import get_runtime_preinstalls
+from lithops.storage.storage import InternalStorage
+from lithops.constants import JOBS_PREFIX
+
+
 
 logger = logging.getLogger('lithops.worker')
 
 
 def main(event, context):
     logger.info("Starting GCP Functions function execution")
+
+    # pub/sub event data is b64 encoded
     args = json.loads(base64.b64decode(event['data']).decode('utf-8'))
-    os.environ['__PW_ACTIVATION_ID'] = uuid.uuid4().hex
-    setup_lithops_logger(args['log_level'])
-    if 'remote_invoker' in args:
-        logger.info("Lithops v{} - Starting invoker".format(__version__))
+    os.environ['__LITHOPS_ACTIVATION_ID'] = uuid.uuid4().hex
+    setup_lithops_logger(args.get('log_level', 'INFO'))
+
+    if 'get_preinstalls' in args:
+        logger.info("Lithops v{} - Generating metadata".format(__version__))
+        internal_storage = InternalStorage(args['get_preinstalls']['storage_config'])
+        object_key = '/'.join([JOBS_PREFIX, args['get_preinstalls']['runtime_name'] + '.meta'])
+        logger.info("Runtime metadata key {}".format(object_key))
+        runtime_meta = get_runtime_preinstalls()
+        runtime_meta_json = json.dumps(runtime_meta)
+        internal_storage.put_data(object_key, runtime_meta_json)
+    elif 'remote_invoker' in args:
+        logger.info("Lithops v{} - Starting IBM CF invoker".format(__version__))
         function_invoker(args)
     else:
-        logger.info("Lithops v{} - Starting execution".format(__version__))
+        logger.info("Lithops v{} - Starting IBM CF execution".format(__version__))
```

Comment:
@aitorarjona IBM CF?

Reply:
sorry, copy-pasted from ibm cf backend, changed it